opencti-platform/opencti-graphql/src/modules/decayRule/decayRule-domain.ts

```diff
@@ -336,12 +336,14 @@ export const initDecayRules = async (context: AuthContext, user: AuthUser) => {
 
 // end region
 export const selectDecayRuleForIndicator = (indicatorObservableType: string, decayRules: DecayRuleConfiguration[]) => {
-  const orderedRules = [...decayRules].sort((a, b) => (b.order || 0) - (a.order || 0));
+  const orderedRules = [...decayRules].filter((d) => d.active)
+    .sort((a, b) => (b.order || 0) - (a.order || 0));
   const decayRule = orderedRules.find((rule) => {
-    if (!indicatorObservableType) {
-      return rule.decay_observable_types?.length === 0;
+    if (rule.decay_observable_types?.length > 0) {
+      // return first rule matching the indicator main observable type
+      return indicatorObservableType && rule.decay_observable_types.includes(indicatorObservableType);
     }
-    return rule.decay_observable_types?.includes(indicatorObservableType);
+    return true; // return first rule
```

Comment:
Now that rules applying to all types are always contained in decayRule, we probably don't need to check for it again on line 351, and we can always return the fallback decay if we found nothing with the find

Reply:
ah right

Reply:
done